File: frontend/src/types/trip.ts
export interface Trip {
  id:                number;
  destination:       string;
  days:              number;
  budget:            number;
  category:          string;
  daily_budget:      number;
  travel_style:      string | null;
  ai_recommendation: string | null;
}

export interface CreateTripPayload {
  destination:  string;
  days:         number;
  budget:       number;
  travel_style: string | null;
}

export interface GenerateTripPayload extends CreateTripPayload {
  travel_style: string;
}

export interface GetTripsParams {
  destination?: string;
  page?:        number;
  limit?:       number;
}

export interface PaginatedTripsResponse {
  total: number;
  page:  number;
  limit: number;
  items: Trip[];
}

File: backend/models/trip.py
from sqlalchemy import Column, Integer, String, Float, Text, DateTime, ForeignKey, func
from sqlalchemy.orm import relationship
from database import Base


class Trip(Base):
    __tablename__ = "trips"

    id                = Column(Integer, primary_key=True, index=True)
    user_id           = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    destination       = Column(String,  nullable=False)
    days              = Column(Integer, nullable=False)
    budget            = Column(Float,   nullable=False)
    category          = Column(String,  nullable=False)
    daily_budget      = Column(Float,   nullable=False)
    travel_style      = Column(String,  nullable=True)
    ai_recommendation = Column(Text,    nullable=True)
    created_at        = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at        = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)
    deleted_at        = Column(DateTime(timezone=True), nullable=True)

    user = relationship("User", back_populates="trips")
